File: facebook_uploader.py
"""
facebook_uploader.py
Facebook Graph API দিয়ে Page-এ ভিডিও পোস্ট করে।
"""
import requests

GRAPH_VIDEO_URL = "https://graph-video.facebook.com/v19.0"


def upload_video(page_id, page_access_token, file_path, description=""):
    if not page_id or not page_access_token:
        raise ValueError("Facebook Page ID / Access Token খালি আছে।")

    url = f"{GRAPH_VIDEO_URL}/{page_id}/videos"
    with open(file_path, "rb") as f:
        files = {"source": f}
        data = {"description": description, "access_token": page_access_token}
        resp = requests.post(url, files=files, data=data, timeout=900)

    if resp.status_code != 200:
        raise RuntimeError(f"Facebook upload failed: {resp.text}")
    return resp.json().get("id", "")

File: app/src/main/java/com/autoreel/app/FacebookUploader.kt
package com.autoreel.app

import okhttp3.MediaType.Companion.toMediaType
import okhttp3.MultipartBody
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.asRequestBody
import org.json.JSONObject
import java.io.File
import java.util.concurrent.TimeUnit

class FacebookUploader(
    private val pageId: String,
    private val pageAccessToken: String
) {
    private val client = OkHttpClient.Builder()
        .connectTimeout(60, TimeUnit.SECONDS)
        .writeTimeout(10, TimeUnit.MINUTES)
        .readTimeout(10, TimeUnit.MINUTES)
        .build()

    fun upload(videoFile: File, description: String): String {
        val requestBody = MultipartBody.Builder()
            .setType(MultipartBody.FORM)
            .addFormDataPart("access_token", pageAccessToken)
            .addFormDataPart("description", description)
            .addFormDataPart("source", videoFile.name, videoFile.asRequestBody("video/mp4".toMediaType()))
            .build()

        val req = Request.Builder()
            .url("https://graph-video.facebook.com/v19.0/$pageId/videos")
            .post(requestBody)
            .build()

        client.newCall(req).execute().use { resp ->
            val text = resp.body?.string() ?: ""
            if (!resp.isSuccessful) throw RuntimeException("Facebook upload failed: $text")
            return JSONObject(text).optString("id", "unknown")
        }
    }
}
